lib/Sema/CSApply.cpp

```diff
@@ -4615,8 +4615,10 @@ namespace {
         conformances.push_back(hashableConformance);
       }
 
-      component.setSubscriptIndexHashableConformances(conformances);
-      return component;
+      return KeyPathExpr::Component::forSubscriptWithPrebuiltIndexExpr(
+          ref, newIndexExpr, cs.getASTContext().AllocateCopy(newLabels),
```

Comment:
It looks like KeyPathExpr::Component::getSubscriptLabels() is only called for KeyPathExpr::Component::Kind::UnresolvedSubscript. It might be worth a try to stick an assertion there if the kind is anything else, and remove the parameter from forSubscriptWithPrebuiltIndexExpr() altogether.